docs/source/guide/resources.rst

```diff
@@ -185,3 +185,18 @@ keyword arguments. Examples of waiters include::
 
     # EC2: Wait for an instance to reach the running state.
     instance.wait_until_running()
+
+
+Multithreading
+--------------
+It is recommended to create a resource instance for each thread in a multithreaded application rather than sharing a single instance among the threads. For example::
+
+    import boto3
+    import threading
+
+    class MyTask(threading.Thread):
+        def run(self):
+            s3 = boto3.resource('s3')
```

Comment:
I'd prefer to move the resource creation out of the thread code, since resource creation isn't guaranteed to be thread safe if you're pulling from the same default session.  Either that or update this to also use separate sessions.